shellfoundry_traffic/script_utils.py

```diff
@@ -18,15 +18,24 @@
 
 class ScriptCommandExecutor:
 
-    def __init__(self, script_definition_yaml: str) -> None:
-        shell_definition_yaml = Path(os.getcwd()).joinpath(f'{script_definition_yaml}.yaml')
-        with open(shell_definition_yaml, 'r') as file:
+    def __init__(self, script_definition: str) -> None:
+        script_definition_yaml = (script_definition if script_definition.endswith('.yaml') else
+                                  f'{script_definition}.yaml')
+        script_definition_yaml_full_path = Path(os.getcwd()).joinpath(script_definition_yaml)
+        with open(script_definition_yaml_full_path, 'r') as file:
             self.script_definition = yaml.safe_load(file)
         self.dist = Path(os.getcwd()).joinpath('dist')
         self.script_zip = self.dist.joinpath(f'{self.script_definition["metadata"]["script_name"]}.zip')
 
     def should_zip(self, file: str) -> bool:
-        return file not in self.script_definition['files']['exclude']
+        """ Returns whether the file should be added to the shell zip file or not. """
+        if file not in self.script_definition['files']['exclude']:
+            return False
```

Comment:
If 'files' is not a key in the self.script_definition dictionary this command will raise an exception.
First you have to check if 'files' entry exists at all.
Then, move on to check the next condition, and I don't think you need a loop for this.

Reply:
Thanks, code changed and checked on all 3 situations in debug mode.